Source/WebCore/rendering/style/StyleCrossfadeImage.cpp

```diff
@@ -152,7 +153,19 @@ RefPtr<Image> StyleCrossfadeImage::image(const RenderElement* renderer, const Fl
     if (!fromImage || !toImage)
         return &Image::nullImage();
 
-    return CrossfadeGeneratedImage::create(*fromImage, *toImage, m_percentage, fixedSize(*renderer), size);
+    RefPtr fromImageRefernce = fromImage;
+    RefPtr toImageReference = toImage;
+
+    if (fromImage->isSVGImage()) {
+        URL fromURL = m_cachedFromImage ? m_cachedFromImage->url() : URL();
+        fromImageRefernce = SVGImageForContainer::create(&downcast<SVGImage>(*fromImage), size, 1, fromURL);
+    }
```

Comment:
You can use `dynamicDowncast` here and in the similar fragment below. I don't think you need the `URL` type qualifier either.

```suggestion
    if (RefPtr fromSVGImage = dynamicDowncast<SVGImage>(protectedFromImage)) {
        auto fromURL = m_cachedFromImage ? m_cachedFromImage->url() : URL();
        protectedFromImage = SVGImageForContainer::create(*fromSVGImage, size, 1, fromURL);
    }
```

Reply:
Need to slightly change to make it compile but updated.